### DISCUSSION TITLE
How to clone filter object? (for cursor based pagination)

### DISCUSSION THREAD

**Discussion by 72L** [supabase/supabase#8718] [General]
I'm using a cursor on id to paginate like this:
```
const query = supabase.from(TABLE).select().order('id')
let thisPageQuery = query;
[loop] {
      thisPageQuery = query.gt('id', lastRow.id)
 ...
}
```
This works. However, I look at my database logs and it turns out the "gt" parameters are stacking inside `thisPageQuery` such that the query looks like this:
```
SELECT * FROM table 
WHERE id > $1 
AND  id > $2 
AND  id > $3 
AND  id > $4 
AND ...
```

How can I clone the initial `query` object so that `thisPageQuery` gets a deep copy?

**Comment by GaryAustin1** [comment]
Easiest thing to do is just move the entire supabase call inside the loop and just change the last id.
